compiler/rustc_abi/src/extern_abi.rs

```diff
@@ -249,6 +249,47 @@ impl ExternAbi {
             _ => false,
         }
     }
+
+    /// Can a function with this ABI be called with a rust call expression?
+    ///
+    /// Some ABIs cannot be called from rust, either because rust does not know how to
+    /// generate code for the call, or because a call does not semantically make sense.
+    pub fn can_be_called_with_call_expr(self) -> bool {
+        match self {
+            // Rust doesn't know how to call functions with this ABI.
+            ExternAbi::Custom => false,
+
+            // This is a function for the GPU, and cannot be called by the host.
+            ExternAbi::PtxKernel | ExternAbi::GpuKernel => false,
```

Comment:
This seems backward? This is a function *for entry from the host*, it cannot be called by the GPU because it's equivalent to a `_start` symbol.